index.js

```diff
@@ -789,68 +763,68 @@ class LatticeKeyring extends EventEmitter {
     return false;
   }
 
-  // Get the chainId for whatever object this is.
-  // Returns a hex string without the 0x prefix
-  _getEthereumJsChainId(tx) {
-    if (typeof tx.getChainId === 'function')
-      return tx.getChainId();
-    else if (tx.common && typeof tx.common.chainIdBN === 'function')
-      return tx.common.chainIdBN().toString(16);
-    else if (typeof tx.chainId === 'number')
-      return tx.chainId.toString(16);
-    else if (typeof tx.chainId === 'string')
-      return tx.chainId;
-    return '1';
-  }
-
   _getCurrentWalletUID() {
-    return this.walletUID || null;
-  }
-
-  // The SDK has an auto-retry mechanism for all requests if a "wrong wallet"
-  // error gets thrown. In such a case, the SDK will re-connect to the device to
-  // find the new wallet UID and will then save that UID to memory and will retry
-  // the original request with that new wallet UID.
-  // Therefore, we should sync the walletUID with `this.walletUID` after each
-  // SDK request. This is a synchronous and fast operation.
-  _syncCurrentWalletUID() {
     if (!this.sdkSession) {
       return null;
     }
     const activeWallet = this.sdkSession.getActiveWallet();
     if (!activeWallet || !activeWallet.uid) {
       return null;
     }
-    const newUID = activeWallet.uid.toString('hex');
-    // If we fetched a walletUID that does not match our current one,
-    // reset accounts and update the known UID
-    if (newUID != this.walletUID) {
-      this.walletUID = newUID;
-    }
-    return this.walletUID;
+    return activeWallet.uid.toString('hex');
   }
+}
 
-  // Make sure we have an SDK connection and, therefore, an active wallet UID.
-  // If we do not, force an unlock, which adds ~5 seconds to the request.
-  _ensureCurrentWalletUID() {
-    return new Promise((resolve, reject) => {
-      if (!!this._getCurrentWalletUID()) {
-        return resolve();
-      }
-      this.unlock()
-      .then(() => {
-        if (!!this._getCurrentWalletUID()) {
-          return resolve()
-        } else {
-          return reject('Could not access Lattice wallet. Please re-connect.')
-        }
-      })
-      .catch((err) => {
-        return reject(err);
-      })
-    })
-  }
+// Get the `v` component of the signature as well as an `initV`
+// parameter, which is what you need to use to re-create an @ethereumjs/tx
+// object. There is a lot of tech debt in @ethereumjs/tx which also
+// inherits the tech debt of ethereumjs-util.
+// *  The legacy `Transaction` type can call `_processSignature` with the regular
+//    `v` value.
+// *  Newer transaction types such as `FeeMarketEIP1559Transaction` will subtract
+//    27 from the `v` that gets passed in, so we need to add `27` to create `initV`
```

Comment:
This comment is incorrect. `FeeMarketEIP1559Transaction` types require `v` to equal 0 or 1 in the constructor: https://github.com/ethereumjs/ethereumjs-monorepo/blob/7330c4ace495903748c606a47a8b993812504db8/packages/tx/src/eip1559Transaction.ts#L226